setup.php

```diff
@@ -75,39 +43,81 @@
         $configured = false;
     }
 
-	$file_written = false;
+?>
 
-    if(is_writeable($configfile) && !$configured){
+<?php if($configured): //TODO, should this reveal exactly how to reconfigure? I don't this it would be a security issue ?>
+    <h2>System already configured</h2>
+<?php else: ?>
 
-		$handle = fopen($configfile, 'w');
+    <?php if(!function_exists('random_bytes') && !function_exists('openssl_random_pseudo_bytes')): ?>
```

Comment:
The existence of `openssl_random_pseudo_bytes` is no guarantee that Selfauth is able to generate a cryptographically secure key!

[The function](https://secure.php.net/manual/en/function.openssl-random-pseudo-bytes.php) can set a `$crypto_strong` boolean which is the only way to know whether OpenSSL did its job right. We can’t test for this before the key generation.